components/src/components/flinkey-service-assigner/flinkey-service-modal/flinkey-modal.tsx

```diff
@@ -0,0 +1,143 @@
+import { Component, Event, EventEmitter, h, Prop, Listen, State } from '@stencil/core';
+import { httpDelete, httpPut, HttpResponse } from '../../../utils/utils';
+
+@Component({
+  tag: 'flinkey-modal',
+  styleUrl: '../../../utils/common.css',
+  shadow: true,
+})
+export class FlinkeyModal {
+  // Customization of modal
+  @Prop() modalTitle: string = '';
+  @Prop() body: string = '';
+
+  // Inherited props
+  @Prop() product: number;
```

Comment:
Why are these props? (`product`, `service`, `unlinkedServices`)

Reply:
For unlink:
We pass "Product" and "Service" as props, from the Table to the modal, in order to specifically select a product to unlink from a service. 

We do not have a global state manager, so we need to pass down the latest data with props.

For link:
unlinkedServices has been removed based on your feedback yesterday. 
Now, we fetch the "unlinkedServices" on the dropdown directly, but, we still need to know which are active services, in order to filter them with OData and get only the services which are not linked.

Reply:
Please read threw the definition of the `Prop` decorator [here](https://stenciljs.com/docs/properties).

Reply:
"... **Props should be used to pass data down from the parent to the child**. Components need to explicitly declare the Props they expect to receive using the @Prop() decorator. Props can be a number, string, boolean, or even an Object or Array."

Product and Service are values from the parent component, which are passed down to its children (the modal component). So the modal component, needs to specify which props it's expected to receive, and therefore, use the prop decorator.

Sorry, but I might not be understanding what you mean here. Would you be so kind to elaborate, please?